### DISCUSSION TITLE
Machinaris security questions

### DISCUSSION THREAD

**Discussion by fisherwei** [guydavis/machinaris#936] [General]
1. add secretkey to port 8927
2. force https between worker/fullnode, letsencrypt cert?
3. auth for machinaris web

**Comment by guydavis** [comment]
Good suggestions.  For now, users have supported these using their own perferred tools such as VPNs, SSH tunnels, and [authentication mechanisms](https://github.com/guydavis/machinaris/wiki/FAQ#why-doesnt-machinaris-offer-an-authentication-mechanism-to-allow-logins).

**Reply by fisherwei** [reply]
LAN is not secure.
Typically, there is not a dedicated VLAN configured on a home network for server.
If my PC is infected with a virus, the LAN may be scanned.

It would be better to add a simple key to the API.
